mne/time_frequency/tfr.py

```diff
@@ -459,111 +680,245 @@ def single_trial_power(data, sfreq, frequencies, use_fft=True, n_cycles=7,
+# TFR(s) class
 
-    # crop data
-    data = data[:, ifmin:ifmax, itmin:itmax]
+class _BaseTFR(ContainsMixin, UpdateChannelsMixin):
+    @property
+    def ch_names(self):
+        return self.info['ch_names']
 
-    times *= 1e3
-    if dB:
-        data = 10 * np.log10((data * data.conj()).real)
+    def crop(self, tmin=None, tmax=None):
+        """Crop data to a given time interval
 
-    vmin, vmax = _setup_vmin_vmax(data, vmin, vmax)
-    return data, times, freqs, vmin, vmax
+        Parameters
+        ----------
+        tmin : float | None
+            Start time of selection in seconds.
+        tmax : float | None
+            End time of selection in seconds.
 
+        Returns
+        -------
+        inst : instance of AverageTFR
+            The modified instance.
+        """
+        mask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
+        self.times = self.times[mask]
+        self.data = self.data[..., mask]
+        return self
 
-class AverageTFR(ContainsMixin, UpdateChannelsMixin):
-    """Container for Time-Frequency data
+    def copy(self):
+        """Return a copy of the instance."""
+        return deepcopy(self)
 
-    Can for example store induced power at sensor level or intertrial
-    coherence.
+    @verbose
+    def apply_baseline(self, baseline, mode='mean', verbose=None):
+        """Baseline correct the data
 
-    Parameters
-    ----------
+        Parameters
+        ----------
+        baseline : tuple or list of length 2
+            The time interval to apply rescaling / baseline correction.
+            If None do not apply it. If baseline is (a, b)
+            the interval is between "a (s)" and "b (s)".
+            If a is None the beginning of the data is used
+            and if b is None then b is set to the end of the interval.
+            If baseline is equal to (None, None) all the time
+            interval is used.
+        mode : None | 'ratio' | 'zscore' | 'mean' | 'percent' | 'logratio' | 'zlogratio'
+            Do baseline correction with ratio (power is divided by mean
+            power during baseline) or zscore (power is divided by standard
+            deviation of power during baseline after subtracting the mean,
+            power = [power - mean(power_baseline)] / std(power_baseline)),
+            mean simply subtracts the mean power, percent is the same as
+            applying ratio then mean, logratio is the same as mean but then
+            rendered in log-scale, zlogratio is the same as zscore but data
+            is rendered in log-scale first.
+            If None no baseline correction is applied.
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+        """  # noqa
+        self.data = rescale(self.data, self.times, baseline, mode,
+                            copy=False)
+
+
+class AverageTFR(_BaseTFR):
+    """Container for Time-Frequency data
```

Comment:
data. (the dot is not systematic in pep8, but I don't see any reason why we don't have one.)